mattermost-plugin/server/plugin.go

```diff
@@ -103,7 +113,21 @@ func (p *Plugin) OnActivate() error {
 		EnableLocalMode:         false,
 		LocalModeSocketLocation: "",
 		AuthMode:                "mattermost",
-	}, "", logger)
+	}
+	var db store.Store
+	db, err = sqlstore.New(cfg.DBType, "", cfg.DBTablePrefix, logger, sqlDB)
+	if err != nil {
+		return fmt.Errorf("error initializing the DB: %v", err)
+	}
+	if cfg.AuthMode == server.MattermostAuthMod {
+		layeredStore, err2 := mattermostauthlayer.New(cfg.DBType, cfg.DBConfigString, db)
+		if err2 != nil {
+			return fmt.Errorf("error initializing the DB: %v", err2)
+		}
+		db = layeredStore
+	}
```

Comment:
This part is getting repeated. I think we can combine the `sqlstore.New` and `mattermostauthlayer.New` inside a single store constructor `store.New`. LMK if you want me to do that in this PR itself.

Reply:
Yes that can work, I kinda prefer the manual approach, because is explicit that the store layers doesn't need to have an specific order, and you can enable/disable them bases on certain logic. But I'm not against that approach, maybe something like `NewStore`? at the server package?

Reply:
Yep, did exactly that!